### PR TITLE
SONiC FRR Upgrade Process Proposal 

### PR DESCRIPTION
**Pull request by adyeung**
SONiC FRR upgrade process proposal for TSC

### CODE REVIEW

File: tsc/frr/sonic_frr_upgrade_process.md
Comment (dgsudharsan) [inline tsc/frr/sonic_frr_upgrade_process.md:31]: We need to define the list of mandatory tests that need to pass in order to confirm FRR is qualified. Need to ensure that the mandatory tests can be run by all vendors. Spytest is not adopted by all vendors and hence should be optional

File: tsc/frr/sonic_frr_upgrade_process.md
Comment (dgsudharsan) [inline tsc/frr/sonic_frr_upgrade_process.md:36]: How do we deal with new feature integration that might require FRR upgrade?  Depending on the size of feature some features can be taken as patches but some may not. Who takes the responsibility of upgrading FRR in such scenarios? Is it the maintainer or the one who is integrating the new feature?

File: tsc/frr/sonic_frr_upgrade_process.md
Comment (dgsudharsan) [inline tsc/frr/sonic_frr_upgrade_process.md:38]: Shouldn't it be the responsibility of FRR community to fix its issues rather than SONiC community working on fixes there? I believe the flow should be SONiC community members should raise the FRR issues in FRR GitHub found during SONiC testing but the responsibility to fix it should be in FRR community. Not all companies have resources with FRR expertise and hence this may be difficult for them to accomplish if they own FRR maintenance

File: tsc/frr/sonic_frr_upgrade_process.md
Comment (dgsudharsan) [inline tsc/frr/sonic_frr_upgrade_process.md:53]: How to determine if an issue fixed is critical to SONiC or not. SONiC doesn't use all the flows in FRR. SONiC should only port the fixes for the FRR issues that are found during testing of SONiC

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: tsc/frr/sonic_frr_upgrade_process.md
Comment (adyeung) [reply tsc/frr/sonic_frr_upgrade_process.md:36]: The proposed upgrade cadence is every 12months during Nov release, the maintainer service term is 12 months. If there is a need for May release upgrade due to feature requirement or other reasons, it should be requested in the May release plan and have the same maintainer to handle the second upgrade.

File: tsc/frr/sonic_frr_upgrade_process.md
Comment (adyeung) [reply tsc/frr/sonic_frr_upgrade_process.md:38]: FRR issues found in SONiC integration maybe specific to SONiC use case, I can change the language not to mandate the maintainer to fix, eg, the fix can come from SONiC contributors or from FRR community, but the maintainer is responsible to drive the fix to unblock SONiC community

File: tsc/frr/sonic_frr_upgrade_process.md
Comment (adyeung) [reply tsc/frr/sonic_frr_upgrade_process.md:53]: I am ok to revise this to port fixes found in SONiC testing if there is no better way to define critical patch

File: tsc/frr/sonic_frr_upgrade_process.md
Comment (adyeung) [reply tsc/frr/sonic_frr_upgrade_process.md:31]: The intent is to add test coverage to compliment the build test, the dilemma is any new tests should also be part of the build test. I am open to suggestions whether there is a need for FRR acceptance tests.

**Comment by lguohan** [comment]
@adyeung , when generating pull request, it is better to first create the branch in your own forked repo. please take a look at this instructions. https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request

**Comment by adyeung** [comment]
Resubmit a PR from a local branch fork, incorporated comments in the new PR 

https://github.com/sonic-net/SONiC/pull/1438

**Pull request closed without merging**
